subproject1/code/chi-square.R

```diff
@@ -1,23 +1,47 @@
 pacman::p_load(
   ggplot2
-)# This code can check whether you have the desired package and load the package after installing.
+) # This code can check whether you have the desired package and load the package after installing.
 
 here::i_am(
   "subproject1/code/chi-square.R"
 )
 
+data_clean <- readRDS(
+  here::here("data/data_clean.rds") 
+) # read the cleaned data
+
+data_ctg <- subset(data_clean, 
+                   select = c(PNEUMONIA, COPD, DIABETES, CARDIOVASCULAR, RENAL_CHRONIC, ICU)
+                   ) # keep only the wanted categorical variables
+
+sum_table <- data.frame(matrix(ncol = 4, nrow = 0)) # empty dataframe to store Chi-square test outputs
+
+for (i in 1:(ncol(data_ctg)-1)) {
```

Comment:
for (i in 1:(ncol(data_ctg)-1)) {
  chisq_result <- chisq.test(data_ctg$ICU, data_ctg[[i]])
  output <- c(names(data_ctg)[i], round(chisq_result$statistic, 4), chisq_result$parameter, round(chisq_result$p.value,4))
  sum_table <- rbind(sum_table, output)
} # for loop that conducts Chi-square tests